build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java

```diff
@@ -0,0 +1,179 @@
+import org.jfrog.build.api.Build;
+import org.jfrog.build.api.builder.ArtifactBuilder;
+import org.jfrog.build.api.util.FileChecksumCalculator;
+import org.jfrog.build.api.util.Log;
+import org.jfrog.build.client.ArtifactoryUploadResponse;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryBuildInfoClientBuilder;
+import org.jfrog.build.extractor.clientConfiguration.client.ArtifactoryBuildInfoClient;
+import org.jfrog.build.extractor.clientConfiguration.deploy.DeployDetails;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import java.io.FileOutputStream;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@SuppressWarnings({"unused", "WeakerAccess"})
+public class GoPublish extends GoCommand {
+
+    private static final String LOCAL_TMP_ZIP_FILENAME = "tmp_package.zip";
+    private static final String LOCAL_DEPLOYABLE_ZIP_FILENAME = "package.zip";
+    private static final String LOCAL_INFO_FILENAME = "package.info";
+    private static final String GO_VERSION_PREFIX = "v";
+    private static final String PKG_ZIP_FILE_EXTENSION = "zip";
+    private static final String PKG_MOD_FILE_EXTENSION = "mod";
+    private static final String PKG_INFO_FILE_EXTENSION = "info";
+
+    private ArrayListMultimap<String, String> properties;
+    private List<Artifact> artifactList = Lists.newArrayList();
+    private String deploymentRepo;
+    private String version;
+
+    /**
+     * Publish go package.
+     *
+     * @param clientBuilder - Client builder for deployment.
+     * @param properties    - Properties to set on each deployed artifact (Build name, Build number, etc...).
+     * @param repo          - Artifactory's repository for deployment.
+     * @param path          - Path to directory contains go.mod.
+     * @param version       - The package's version.
+     * @param logger        - The logger.
+     */
+    public GoPublish(ArtifactoryBuildInfoClientBuilder clientBuilder, ArrayListMultimap<String, String> properties, String repo, Path path, String version, Log logger) throws  IOException {
+        super(clientBuilder, path, logger);
+        this.properties = properties;
+        this.deploymentRepo = repo;
+        this.version = GO_VERSION_PREFIX + version;
+    }
+
+    public Build execute() {
+        try (ArtifactoryBuildInfoClient dependenciesClient = (ArtifactoryBuildInfoClient) clientBuilder.build()) {
+            client = dependenciesClient;
+            preparePrerequisites(deploymentRepo, client);
+            publishPkg();
+            return createBuild(artifactList, null);
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    /*
+     * Go pkg contains 3 files:
```

Comment:
Go pkg contains 3 files: --> The deployment of a go package requires 3 files:

(the go package is actually the zip).